Scriptrun

hiep-tk opened the pull request:
**What this PR does**:

**Why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

t-kikuc commented on pkg/app/pipedv1/plugin/scriptrun/main.go:
```diff
@@ -0,0 +1,18 @@
+)
+
+func main() {
+	plugin, err := sdk.NewPlugin("scriptrun", "0.0.1", sdk.WithStagePlugin(&plugin{}))
```
`scriptrun-stage` is better like `wait-stage`

Sorry, the name convention is not determined yet.

t-kikuc commented on pkg/app/pipedv1/plugin/scriptrun/options.go:
```diff
@@ -0,0 +1,26 @@
+	config "github.com/pipe-cd/pipecd/pkg/configv1"
+)
+
+type ScriptRunStageOptions struct {
```
make it private (we'll fix waitStageOptions)

t-kikuc commented on pkg/app/pipedv1/plugin/scriptrun/options.go:
```diff
@@ -0,0 +1,26 @@
+	if err := json.Unmarshal(data, &opts); err != nil {
+		return ScriptRunStageOptions{}, fmt.Errorf("failed to unmarshal the config: %w", err)
+	}
+	if opts.Timeout <= 0 {
```
split into validate() method

t-kikuc commented on pkg/app/pipedv1/plugin/scriptrun/plugin.go:
```diff
@@ -0,0 +1,144 @@
+	"time"
+)
+
+const stageScriptRun = "SCRIPT_RUN"
```
maybe need scriptrun rollback stage

t-kikuc commented on pkg/app/pipedv1/plugin/scriptrun/plugin.go:
```diff
@@ -0,0 +1,144 @@
+		Stages: stages,
+	}, nil
+}
+func (p *plugin) ExecuteStage(ctx context.Context, _ sdk.ConfigNone, _ sdk.DeployTargetsNone, input *sdk.ExecuteStageInput[struct{}]) (*sdk.ExecuteStageResponse, error) {
```
creata executeStage() and add tests

t-kikuc commented on pkg/app/pipedv1/plugin/scriptrun/plugin.go:
```diff
@@ -0,0 +1,144 @@
+		}
+		stages = append(stages, stage)
+	}
+
```
add rollback later (or add TODO comment for now)

t-kikuc commented on pkg/app/pipedv1/plugin/scriptrun/plugin.go:
```diff
@@ -0,0 +1,144 @@
+		} else {
+			c <- sdk.StageStatusSuccess
+		}
+	}()
```
split into func(s) and add test

Warashi committed:
Remove check workflow & improve DCO check script (#5904)

* Remove non-maintainer check workflow

This workflow is a bit complex compared to it's purpose

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Fix the commit message check to ignore merge commits

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

---------

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

hiep-tk committed:
add more fields from deployment proto to deployment model for scriptrun plugin (#5902)

Signed-off-by: hiep-tk <hiep.trinhkhanh0466@gmail.com>

github-actions[bot] committed:
[bot] Update contributors (#5906)

Signed-off-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: t-kikuc <97105818+t-kikuc@users.noreply.github.com>

Warashi committed:
Revert "Do treeless clone for git clone to improve performance (#5722)" (#5908)

This reverts commit 7e74937ab808d803adae7c3cd05402b0094d203c.

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

t-kikuc committed:
Add 'title' to distinguish multiple projects in actions-plan-preview (#5905)

* add title to args

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* Add the title to filter latest comments

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* Use hash

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* Fix tests

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* add multibyte test

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

---------

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

github-actions[bot] committed:
Cut release v0.52.1 (#5909)

Signed-off-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Warashi <3600530+Warashi@users.noreply.github.com>

github-actions[bot] committed:
[bot] Publish quickstart manifests (#5912)

Signed-off-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Warashi <3600530+Warashi@users.noreply.github.com>

ffjlabo committed:
[K8s Plugin] Add helper functions for canary stage (#5892)

* Add helper functions for canary stage

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Add patchManifests function

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix for lint

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

---------

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

Warashi committed:
[K8s plugin] Add helper function to delete all variant resources (#5903)

* Add workload and service identification methods to Manifest struct

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add deleteVariantResources function to manage variant resource deletion

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Pre allocate slices to improve performance

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Fix the workload selection to match with the previous implementation

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add unit test for IsService

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add unit test for IsWorkload

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add t.Parallel() to manifest_test.go

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add t.Parallel() to all tests in manifest_test.go

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

---------

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

Warashi committed:
[K8s plugin] implement K8S_BASELINE_CLEAN stage (#5913)

* Fix argument type of deleteVariantResources

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Implement K8S_BASELINE_CLEAN stage

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add testdata for K8S_BASELINE_CLEAN stage

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add test for K8S_BASELINE_CLEAN

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Fix deleteVariantResources to add kubeconfig path as its argument

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Add test for K8S_BASELINE_CLEAN without createService

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

---------

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

ffjlabo committed:
[K8s plugin] Implement caanry stage for k8s plugin (#5914)

* Implement caanry stage for k8s plugin

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix for lint

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

---------

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

ffjlabo committed:
[K8s plugin] Fix to get all resources without using 'kubectl get all' (#5917)

* Fix to get all resources without using 'kubectl get all'

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Add comment

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Implement same logic to k8s multicluster plugin

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Add test case for no resources found

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

---------

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

ffjlabo committed:
Implement canary clean stage (#5918)

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

Warashi committed:
[K8s plugin] Delete baseline/canary variant resources on K8S_ROLLBACK stage (#5919)

ffjlabo committed:
Return stage status fail when failing applyManifests (#5921)

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

Warashi committed:
Fix nil pointer dereference error on spec.plugins config on app config not set (#5922)

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

khanhtc1202 committed:
Make build plugin command accept GO build flags and envs (#5923)

Signed-off-by: khanhtc1202 <khanhtc1202@gmail.com>

t-kikuc committed:
add area/sdk-plugins (#5924)

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

t-kikuc committed:
add Note of 'Visible' (#5925)

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

Warashi committed:
Fix the build quick sync stages method to return an empty response (#5926)

* Fix the build quick sync stages method to return an empty response

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

* Update pkg/plugin/sdk/deployment.go

Co-authored-by: Khanh Tran <32532742+khanhtc1202@users.noreply.github.com>
Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

---------

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>
Co-authored-by: Khanh Tran <32532742+khanhtc1202@users.noreply.github.com>

dependabot[bot] committed:
Bump webpack-dev-server from 5.2.0 to 5.2.1 in /web (#5927)

Bumps [webpack-dev-server](https://github.com/webpack/webpack-dev-server) from 5.2.0 to 5.2.1.
- [Release notes](https://github.com/webpack/webpack-dev-server/releases)
- [Changelog](https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-server/compare/v5.2.0...v5.2.1)

---
updated-dependencies:
- dependency-name: webpack-dev-server
  dependency-version: 5.2.1
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

t-kikuc committed:
Add default response of DetermineStrategy() of DeploymentPlugin (#5929)

* add default for DetermineStrategy()

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* Clarify the comment

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

---------

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

t-kikuc committed:
pass 'name' from piped to plugin (#5931)

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

khanhtc1202 committed:
fix: upgrade @loadable/component from 5.16.4 to 5.16.7 (#5932)

Snyk has created this PR to upgrade @loadable/component from 5.16.4 to 5.16.7.

See this package in yarn:
@loadable/component

See this project in Snyk:
https://app.snyk.io/org/pipecd/project/f41c5767-b506-4f59-beb9-ef662258eb9a?utm_source=github&utm_medium=referral&page=upgrade-pr

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>
Co-authored-by: snyk-bot <snyk-bot@snyk.io>

t-kikuc committed:
Ignore when DetermineStrategy() is unsupported or returns nil (#5930)

* handle Unsupported

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* prevent nil panic when the response is nil

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

---------

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

ffjlabo committed:
[K8s plugin] Add README for k8s plugin (#5934)

* WIP

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Add overview

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix doc

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Remove pipeline section

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Remove link

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix overview

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix example

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

---------

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

t-kikuc committed:
User PipedPlugin.Name instead of passing --name (#5936)

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

ffjlabo committed:
Update SDK version (#5935)

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

t-kikuc committed:
add usage of Wait Plugin (#5920)

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

ffjlabo committed:
[SDK] Allow users to set nothing config for the plugin in the app.pipecd.yaml (#5937)

* Allaw users to set nothing config for the plugin in the app.pipecd.yaml

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Remove &

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix comment

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix comment

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

---------

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

t-kikuc committed:
Deprecate `ArtifactVersion.kind` (#5939)

* deprecate model.ArtifactVersion.kind (not deleted)

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* removed sdk.ArtifactVersion.Kind

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

---------

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

ffjlabo committed:
Set default empty object to with field (#5943)

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

caniszczyk committed:
fix: upgrade react-router-dom from 6.30.0 to 6.30.1 (#5938)

Snyk has created this PR to upgrade react-router-dom from 6.30.0 to 6.30.1.

See this package in yarn:
react-router-dom

See this project in Snyk:
https://app.snyk.io/org/pipecd/project/f41c5767-b506-4f59-beb9-ef662258eb9a?utm_source=github&utm_medium=referral&page=upgrade-pr

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>
Co-authored-by: snyk-bot <snyk-bot@snyk.io>

ffjlabo committed:
Ignore unsupported plugin when reporting livestate (#5942)

* Ignore upsupported plugin when reporting livestate

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Unique plugin clients

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

---------

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

ffjlabo committed:
Update sdk version (#5945)

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

t-kikuc committed:
Add index validation of buildPipelineSyncStages in planner (#5946)

* add validateStageIndexes in planner

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* add api spec comments

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* Fix: allow duplication of non-rollback and rollback

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

* fix typo

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

---------

Signed-off-by: t-kikuc <tkikuchi07f@gmail.com>

ffjlabo committed:
Add --match opion (#5949)

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

ffjlabo committed:
Fix README-usage (#5944)

* Fix README-usage

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Update README alpha

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Update

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

* Fix as --insecure option

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

---------

Signed-off-by: Yoshiki Fujikane <ffjlabo@gmail.com>

khanhtc1202 committed:
Add blog about alpha release of plugin-arch piped (#5948)

Warashi committed:
Fix version tag in the CI (#5953)

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

sagnik3788 committed:
Remove deprecated  field from Deployment (#5809)

* remove deprecated  field from Deployment

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>

* remove deprecated  field from deploymentstore.go and the test file

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>

* remove deprecated  field

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>

* Fix all lint errors

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>

* add Versions

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>

* remove version arg

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>

* fix web/lint error

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>

---------

Signed-off-by: sagnik3788 <sagnikdas5432@gmail.com>
Signed-off-by: Sagnik Das <116512372+sagnik3788@users.noreply.github.com>
Co-authored-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

golemiso committed:
Add option --dependency-update to command helm template (#5877)

Signed-off-by: golemiso <3282656+golemiso@users.noreply.github.com>
Co-authored-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

hiep-tk committed:
add index to ExecuteStageRequest

Signed-off-by: hiep-tk <hiep.trinhkhanh0466@gmail.com>

hiep-tk committed:
scriptrun ported with rollback implementation

Signed-off-by: hiep-tk <hiep.trinhkhanh0466@gmail.com>

Warashi committed:
Improve docs for piped-plugin-sdk-go (#5957)

Signed-off-by: Shinnosuke Sawada-Dazai <shin@warashi.dev>

khanhtc1202 committed:
Remove pipectl init command (#5958)

Signed-off-by: khanhtc1202 <khanhtc1202@gmail.com>

Warashi committed:
[SDK] add unit types to use in config types (#5959)
